packages/solidstart/src/vite/sentrySolidStartVite.ts

```diff
@@ -1,13 +1,36 @@
-import type { Plugin } from 'vite';
+import type { Plugin, UserConfig } from 'vite';
+import { makeBuildInstrumentationFilePlugin } from './buildInstrumentationFile';
 import { makeSourceMapsVitePlugin } from './sourceMaps';
 import type { SentrySolidStartPluginOptions } from './types';
 
+// todo(v9): Don't export to users anymore and remove deprecation (and eslint warning silencing) when it's not exported anymore
 /**
  * Various Sentry vite plugins to be used for SolidStart.
+ *
+ * @deprecated This plugin will be removed in v9. Instead, use `withSentry` to wrap your SolidStart config. Example:
+ * ```
+ * export default defineConfig(
+ *   withSentry(
+ *     {
+ *       // SolidStart config...
+ *     },
+ *     {
+ *       // Sentry config
+ *       org: process.env.SENTRY_ORG,
+ *       project: process.env.SENTRY_PROJECT,
+ *       authToken: process.env.SENTRY_AUTH_TOKEN,
+ *     },
+ *   ),
+ * );
+ * ```
```

Comment:
Everything in this PR is just a backport of the PRs merged into the v9 branch.

Those JSDoc lines are new. And I kept the E2E tests `solidstart` and `solidstart-spa` as they were before, to still test that the old way of setting up the SDK (with the Vite plugin) works.